task/sensor/src/main.rs

```diff
@@ -69,11 +70,38 @@ impl idl::InOrderSensorImpl for ServerImpl {
 
         if index < NUM_SENSORS {
             self.data[index] = Reading::NoData(nodata);
+
+            //
+            // We pack per-`NoData` counters into a u32.
+            //
+            let (nbits, shift) = nodata.counter_encoding::<u32>();
+            let mask = (1 << nbits) - 1;
+            let bitmask = mask << shift;
+            let incr = 1 << shift;
+
+            if self.nerrors[index] & bitmask != bitmask {
```

Comment:
This isn't obvious at first glance; maybe a comment here explaining that it's checking for saturating?

Reply:
Definitely; thanks, and fixed.